runtime/src/iree/builtins/ukernel/tools/util.c

```diff
@@ -318,13 +229,34 @@ bool iree_uk_cpu_supports(const iree_uk_uint64_t* cpu_data_fields) {
   return true;
 }
 
+static const char* iree_uk_cpu_feature_name(int feature_field_index,
+                                            int feature_bit_pos) {
+  IREE_UK_ASSERT(feature_field_index >= 0 &&
+                 feature_field_index < IREE_CPU_DATA_FIELD_COUNT);
+  IREE_UK_ASSERT(feature_bit_pos >= 0 && feature_bit_pos < 64);
+#define IREE_CPU_FEATURE_BIT(arch, field_index, bit_pos, bit_name, llvm_name) \
+  if (IREE_ARCH_ENUM == IREE_ARCH_ENUM_##arch) {                              \
+    if (field_index == feature_field_index && bit_pos == feature_bit_pos) {   \
+      return llvm_name;                                                       \
+    }                                                                         \
+  }
+#include "iree/schemas/cpu_feature_bits.inl"
+#undef IREE_CPU_FEATURE_BIT
+  IREE_UK_ASSERT(false && "Unknown CPU feature bit");
+  return NULL;
+}
+
 const char* iree_uk_cpu_first_unsupported_feature(
-    const iree_uk_cpu_features_list_t* cpu_features) {
-  for (int i = 0; i < cpu_features->size; ++i) {
-    int64_t supported = 0;
-    IREE_CHECK_OK(iree_cpu_lookup_data_by_key(IREE_SV(cpu_features->entries[i]),
-                                              &supported));
-    if (!supported) return cpu_features->entries[i];
+    const iree_uk_uint64_t* cpu_data_fields) {
+  for (int i = 0; i < IREE_CPU_DATA_FIELD_COUNT; ++i) {
```

Comment:
Is IREE_CPU_DATA_FIELD_COUNT guaranteed to just be a constant? (Else should we do initialize local variable to it? for (int i = 0, e = ...; ...; ++i))